### PR TITLE
[SDK-4039] Make `modulereport` print gzipped size too

### LINKED ISSUE #1580: Make `modulereport` script output the gzipped size of the bundle too

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
It’s quite likely that some of our users will be serving their bundles with gzip encoding, so this is useful information to have.

<details>
<summary>
Example output
</summary>

```
> ably@1.2.48 modulereport
> tsc --noEmit scripts/moduleReport.ts && esr scripts/moduleReport.ts

Minimal useful Realtime (BaseRealtime + FetchRequest + WebSocketTransport): raw: 107.68 KiB; gzip: 31.15 KiB
BaseRest: raw: 59.12 KiB; gzip: 18.52 KiB
BaseRest + Rest: raw: 59.12 KiB; gzip: 18.52 KiB
BaseRest + Crypto: raw: 62.66 KiB; gzip: 19.76 KiB
BaseRest + MsgPack: raw: 67.88 KiB; gzip: 20.62 KiB
BaseRest + RealtimePresence: raw: 86.54 KiB; gzip: 25.73 KiB
BaseRest + XHRPolling: raw: 77.17 KiB; gzip: 23.75 KiB
BaseRest + XHRStreaming: raw: 77.17 KiB; gzip: 23.75 KiB
BaseRest + WebSocketTransport: raw: 72.06 KiB; gzip: 22.08 KiB
BaseRest + XHRRequest: raw: 66.24 KiB; gzip: 20.80 KiB
BaseRest + FetchRequest: raw: 60.16 KiB; gzip: 18.96 KiB
BaseRest + MessageInteractions: raw: 60.36 KiB; gzip: 18.93 KiB
BaseRest + Vcdiff: raw: 66.55 KiB; gzip: 20.63 KiB
BaseRest + generateRandomKey: raw: 62.69 KiB; gzip: 19.77 KiB
BaseRest + getDefaultCryptoParams: raw: 62.69 KiB; gzip: 19.77 KiB
BaseRest + decodeMessage: raw: 59.46 KiB; gzip: 18.63 KiB
BaseRest + decodeEncryptedMessage: raw: 62.99 KiB; gzip: 19.85 KiB
BaseRest + decodeMessages: raw: 59.56 KiB; gzip: 18.65 KiB
BaseRest + decodeEncryptedMessages: raw: 63.10 KiB; gzip: 19.88 KiB
BaseRest + decodePresenceMessage: raw: 59.30 KiB; gzip: 18.56 KiB
BaseRest + decodePresenceMessages: raw: 59.40 KiB; gzip: 18.58 KiB
BaseRest + constructPresenceMessage: raw: 59.13 KiB; gzip: 18.53 KiB
BaseRealtime: raw: 103.10 KiB; gzip: 29.69 KiB
BaseRealtime + Rest: raw: 122.58 KiB; gzip: 34.47 KiB
BaseRealtime + Crypto: raw: 106.64 KiB; gzip: 30.93 KiB
BaseRealtime + MsgPack: raw: 111.86 KiB; gzip: 31.78 KiB
BaseRealtime + RealtimePresence: raw: 113.14 KiB; gzip: 32.19 KiB
BaseRealtime + XHRPolling: raw: 111.61 KiB; gzip: 32.36 KiB
BaseRealtime + XHRStreaming: raw: 111.61 KiB; gzip: 32.36 KiB
BaseRealtime + WebSocketTransport: raw: 106.63 KiB; gzip: 30.70 KiB
BaseRealtime + XHRRequest: raw: 106.83 KiB; gzip: 31.05 KiB
BaseRealtime + FetchRequest: raw: 104.14 KiB; gzip: 30.13 KiB
BaseRealtime + MessageInteractions: raw: 104.34 KiB; gzip: 30.08 KiB
BaseRealtime + Vcdiff: raw: 110.53 KiB; gzip: 31.79 KiB
BaseRealtime + generateRandomKey: raw: 106.67 KiB; gzip: 30.95 KiB
BaseRealtime + getDefaultCryptoParams: raw: 106.67 KiB; gzip: 30.95 KiB
BaseRealtime + decodeMessage: raw: 103.44 KiB; gzip: 29.80 KiB
BaseRealtime + decodeEncryptedMessage: raw: 106.98 KiB; gzip: 31.04 KiB
BaseRealtime + decodeMessages: raw: 103.54 KiB; gzip: 29.82 KiB
BaseRealtime + decodeEncryptedMessages: raw: 107.08 KiB; gzip: 31.06 KiB
BaseRealtime + decodePresenceMessage: raw: 104.60 KiB; gzip: 29.98 KiB
BaseRealtime + decodePresenceMessages: raw: 104.70 KiB; gzip: 30.01 KiB
BaseRealtime + constructPresenceMessage: raw: 104.43 KiB; gzip: 29.95 KiB
generateRandomKey: raw: 38.94 KiB; gzip: 13.60 KiB
generateRandomKey + Crypto: raw: 38.94 KiB; gzip: 13.59 KiB
getDefaultCryptoParams: raw: 38.94 KiB; gzip: 13.59 KiB
getDefaultCryptoParams + Crypto: raw: 38.94 KiB; gzip: 13.59 KiB
decodeMessage: raw: 38.34 KiB; gzip: 13.31 KiB
decodeEncryptedMessage: raw: 41.88 KiB; gzip: 14.56 KiB
decodeEncryptedMessage + Crypto: raw: 41.88 KiB; gzip: 14.56 KiB
decodeMessages: raw: 38.45 KiB; gzip: 13.35 KiB
decodeEncryptedMessages: raw: 41.98 KiB; gzip: 14.59 KiB
decodeEncryptedMessages + Crypto: raw: 41.98 KiB; gzip: 14.58 KiB
decodePresenceMessage: raw: 38.51 KiB; gzip: 13.42 KiB
decodePresenceMessages: raw: 38.61 KiB; gzip: 13.44 KiB
constructPresenceMessage: raw: 36.51 KiB; gzip: 12.68 KiB
```
</details>

Resolves #1580.

### CODE REVIEW

**Comment by lawrence-forooghian** [comment]
Just pushed a couple of other commits, to format as a table and add the size of a bundle that includes _all_ modules (a piece of information that @ttypic indicated would be useful to have).

File: scripts/moduleReport.ts [hunk 1]
```diff
@@ -91,54 +116,57 @@ async function runSourceMapExplorer(bundleInfo: BundleInfo) {
...
-    console.log(`${baseClient}: ${formatBytes(baseClientSize)}`);
+    // First output the size of the base client
+    output.tableRows.push({ description: baseClient, sizes: baseClientSizes });
 
-    // Then display the size of each export together with the base client
-    [...moduleNames, ...Object.values(functions).map((functionData) => functionData.name)].forEach((exportName) => {
-      const size = getImportSize([baseClient, exportName]);
-      console.log(`${baseClient} + ${exportName}: ${formatBytes(size)}`);
+    // Then output the size of each export together with the base client
+    for (const exportName of [...moduleNames, ...Object.values(functions).map((functionData) => functionData.name)]) {
```
Comment (VeskeR) [inline scripts/moduleReport.ts:129]: `Object.values(functions).map` can be just `functions.map`

File: scripts/moduleReport.ts [hunk 2]
```diff
@@ -256,15 +301,33 @@ async function checkBaseRealtimeFiles() {
...
-  const errors: Error[] = [];
-
-  errors.push(...printAndCheckMinimalUsefulRealtimeBundleSize());
-  errors.push(...printAndCheckModuleSizes());
-  errors.push(...printAndCheckFunctionSizes());
-  errors.push(...(await checkBaseRealtimeFiles()));
+  const output = (
+    await Promise.all([
+      calculateAndCheckMinimalUsefulRealtimeBundleSize(),
+      calculateAllModulesBundleSize().then((tableRow): Output => ({ tableRows: [tableRow], errors: [] })),
```
Comment (VeskeR) [inline scripts/moduleReport.ts:307]: let's make `calculateAllModulesBundleSize` return `Output` object directly and remove this .then call for consistency with other calculate functions

**Review by VeskeR — CHANGES_REQUESTED** [review]
LGMT, let's merge after minor changes

**Pull request merged**